subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/fixture/AbstractJvmFailFastIntegrationSpec.groovy

```diff
@@ -0,0 +1,215 @@
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        sleep(1000)
+        testExecution.release(OTHER_RESOURCE)
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        result.testClass('pkg.OtherTest').assertTestCount(0, 0, 0)
+    }
+
+    def "ensure fail fast with forkEvery"() {
+        final testOmitted = 5
+        given:
+        withBuildFile(2)
+        withFailingTest()
+        def otherResources = withNonfailingTests(testOmitted)
+        def testExecution = server.expectConcurrentAndBlock(2, ([ FAILED_RESOURCE ] + otherResources).grep() as String[])
+
+        when:
+        def gradleHandle = executer.withTasks('test', '--fail-fast').start()
+        testExecution.waitForAllPendingCalls()
+
+        then:
+        testExecution.release(FAILED_RESOURCE)
+        testExecution.waitForAllPendingCalls()
+        sleep(1000)
+        testExecution.releaseAll()
+        gradleHandle.waitForFailure()
+        def result = new DefaultTestExecutionResult(testDirectory)
+        result.testClass('pkg.FailedTest').assertTestFailed('failTest', Matchers.anything())
+        testOmitted.each {
+            result.testClass("pkg.OtherTest_${it}").assertTestCount(0, 0, 0)
+        }
+    }
+
+    private void withBuildFile(int forkEvery = 0) {
+        buildFile << """
+            apply plugin: 'java'
+            
+            ${RepoScriptBlockUtil.jcenterRepository()}
+            
+            dependencies {
+                testCompile '${testDependency()}'
+            }
+            
+            tasks.withType(Test) {
+                maxParallelForks = $MAX_WORKERS
+                forkEvery = $forkEvery
+            }
+
+            ${testFrameworkConfiguration()}
+        """
+    }
+
+    private void withFailingTest() {
+        file('src/test/java/pkg/FailedTest.java') << """
+            package pkg;
+            import ${testAnnotationClass()};
+            public class FailedTest {
+                @Test
+                public void failTest() {
+                    ${server.callFromBuild("$FAILED_RESOURCE")}
+                    throw new RuntimeException();
+                }
+            }
+        """.stripIndent()
+    }
+
+    private void withNonfailingTest() {
+        file('src/test/java/pkg/OtherTest.java') << """
+            package pkg;
+            import ${testAnnotationClass()};
+            public class OtherTest {
+                @Test
+                public void passingTest() {
+                    ${server.callFromBuild("$OTHER_RESOURCE")}
+                }
+            }
+        """.stripIndent()
+    }
+
+    private List<String> withNonfailingTests(int num) {
+        (1..num).collect {
```

Comment:
I think [findResults](http://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/Iterable.html#findResults(groovy.lang.Closure)) is what you need here.

Reply:
the modulo is no longer necessary with the addition of `BlockingHttpServer.expectOptionalAndBlock()`